packages/salesforce-adapter/src/client/client.ts

```diff
@@ -264,17 +265,34 @@ const realConnection = (
   })
 )
 
+
+const isSocketTimeoutError = (e: Error): boolean => (
+  e.message === 'ESOCKETTIMEDOUT'
+)
+
+type ConfigSuggestionsInput<T> = {
+  error: Error
+  chunkInput: T
+}
+
+type ConfigSuggestionsCreator<T> = {
+  predicate: (input: ConfigSuggestionsInput<T>) => boolean
+  create: (input: ConfigSuggestionsInput<T>) => ConfigChangeSuggestion[]
```

Comment:
Any reason to keep both methods separate?

```suggestion
  toConfigChangeSuggestion: (input: ConfigSuggestionsInput<T>) => ConfigChangeSuggestion[]
```

And in case it's empty return a [], and in case it's not return the suggestion you want.
Why split to 2 methods?

Reply:
It was the original code and it made the code less readable IMO.
It will require an elvis operator in every `ConfigSuggestionsCreator`. It made the code cleaner to assume that within the context of `create` the validation already passed.